packages/frontend/src/components/Explorer/CustomDimensionModal/CustomBinDimensionModal.tsx

```diff
@@ -59,29 +60,13 @@ export const CustomBinDimensionModal: FC<{
         (context) => context.actions.editCustomDimension,
     );
 
-    const form = useForm({
-        initialValues: {
-            customDimensionLabel: '',
-            binType: BinType.FIXED_NUMBER,
-            binConfig: {
-                fixedNumber: {
-                    binNumber: MIN_OF_FIXED_NUMBER_BINS,
-                },
-                fixedWidth: {
-                    binWidth: MIN_OF_FIXED_NUMBER_BINS,
-                },
-                customRange: DEFAULT_CUSTOM_RANGE,
-            },
-        },
-        validate: {
-            customDimensionLabel: (label) => {
-                if (!label) return null;
-
-                if (!item) return null;
+    const formSchema = z.object({
+        customDimensionLabel: z.string().refine(
```

Comment:
since I moved the validation to zod, I also checked if it works the same way it used to. it does:
![CleanShot 2024-10-17 at 20 35 04@2x](https://github.com/user-attachments/assets/1442312f-c7ef-4db2-b7cc-65e98d8ac97c)